data/battle_scripts_1.s

```diff
@@ -417,13 +417,35 @@ gBattleScriptsForMoveEffects::
 	.4byte BattleScript_EffectSpecialAttackUpHit      @ EFFECT_SPECIAL_ATTACK_UP_HIT
 	.4byte BattleScript_EffectVictoryDance            @ EFFECT_VICTORY_DANCE
 	.4byte BattleScript_EffectTeatime                 @ EFFECT_TEATIME
+	.4byte BattleScript_EffectCeaselessEdge           @ EFFECT_CEASELESS_EDGE
+	.4byte BattleScript_EffectDireClaw                @ EFFECT_DIRE_CLAW
+	.4byte BattleScript_EffectStoneAxe                @ EFFECT_STONE_AXE
+	.4byte BattleScript_EffectBarbBarrage             @ EFFECT_BARB_BARRAGE
+
+BattleScript_EffectStoneAxe::
+	call BattleScript_EffectHit_Ret
```

Comment:
I don't like having a copy of EffectHit with a return. I would suggest making a MOVE_EFFECT_STEALTH_ROCK that can be used here. In addition, it shouldnt set stealth rock if the battle will end (there are similar checks in other MOVE_EFFECT cases)

Reply:
> I don't like having a copy of EffectHit with a return.

Why not? It's far, far better than having what is basically 95% of it copy-pasted :/

I already made use of it for some of the other effects in the upcoming batches, but that aside, I did it *because* it would make situations like this where you have move effects that are mostly copy-pastes of `EFFECT_HIT`'s own battle script and only deviate from it a little by the end easier to handle.

> I would suggest making a MOVE_EFFECT_STEALTH_ROCK that can be used here.

I mean, that sounds fine too, I guess? But is there an actual reason to do that?

> In addition, it shouldnt set stealth rock if the battle will end (there are similar checks in other MOVE_EFFECT cases)

Actually, it should. #JustGameFreakThings

https://user-images.githubusercontent.com/4485172/210582829-5f51a9a8-b274-4c1b-8f7f-21a4900107b9.mp4